src/components/Semester.vue

```diff
@@ -249,6 +249,13 @@ export default {
         .catch(error => {
           console.log('Error getting document:', error);
         });
+    },
+
+    click(course) {
+      // this.$emit('update', {subject: course.subject, number: course.code, roster: 'FA19', color: course.color});
+      this.$parent.$parent.updateBar({
+        subject: course.subject, number: course.code, roster: 'FA19', color: course.color
+      });
```

Comment:
Replaced cases of $parent with $emit for all cases that involve passing data for the bottom bar. Is it appropriate to use $parent when we're trying to call a function in the parent component to change an independent aspect of that component (e.g. this.$parent.closeBar())?

Reply:
The point is that `this.$parent` makes the code non-compositional. A child component should not need to know its parent otherwise we will have cyclic dependencies.

Reply:
We'll change it